src/app/berkeley_migration_verifier/berkeley_migration_verifier.ml

```diff
@@ -0,0 +1,516 @@
+        Sql.Berkeley.get_all_canonical_blocks_till_height db height )
+  in
+  let%bind all_blocks_till_height =
+    query_migrated_db ~f:(fun db ->
+        Sql.Berkeley.get_all_blocks_till_height db height )
+  in
+  if Int.equal all_blocks_till_height canonical_blocks_count then
+    Deferred.return Check.ok
+  else
+    Deferred.return
+      (Check.err
+         (sprintf "Expected to have only %d canonical block while having %d"
+            canonical_blocks_count all_blocks_till_height ) )
+
+let assert_migrated_db_has_account_accessed query_migrated_db =
+  let%bind account_accessed_count =
+    query_migrated_db ~f:(fun db -> Sql.Berkeley.count_account_accessed db)
+  in
+  if Int.( > ) account_accessed_count 0 then Deferred.return Check.ok
+  else
+    Deferred.return
+      (Check.err
+         (sprintf
+            "Expected to have at least one entry in account accessed table" ) )
+
+let assert_all_user_and_interal_commands_account_ids_appear_in_accounts_accessed
+    query_migrated_db =
+  let%bind account_accessed_in_commands_count =
+    query_migrated_db ~f:(fun db ->
+        Sql.Berkeley.get_account_id_accessed_in_commands db )
+  in
+  let%bind account_accessed =
+    query_migrated_db ~f:(fun db -> Sql.Berkeley.count_account_accessed db)
+  in
+  if Int.equal account_accessed account_accessed_in_commands_count then
+    Deferred.return Check.ok
+  else
+    Deferred.return
+      (Check.err
+         (sprintf
+            "Unmigrated accounts in user or internal commands found. Expected: \
+             '%d' unique accounts based on commands tables but got '%d'"
+            account_accessed account_accessed_in_commands_count ) )
+
+let compare_db_content ~mainnet_archive_uri ~migrated_archive_uri
+    ~fork_state_hash () =
+  printf
+    "Running verifications between '%s' and '%s' schemas. It may take a couple \
+     of minutes... \n"
+    mainnet_archive_uri migrated_archive_uri ;
+
+  let mainnet_archive_uri = Uri.of_string mainnet_archive_uri in
+  let migrated_archive_uri = Uri.of_string migrated_archive_uri in
+  let mainnet_pool =
+    Caqti_async.connect_pool ~max_size:128 mainnet_archive_uri
+  in
+  let migrated_pool =
+    Caqti_async.connect_pool ~max_size:128 migrated_archive_uri
+  in
+
+  match (mainnet_pool, migrated_pool) with
+  | Error _e, _ | _, Error _e ->
+      failwithf "Connection failed to orignal and migrated schema" ()
+  | Ok mainnet_pool, Ok migrated_pool ->
+      Test.of_check Check.ok ~name:"Connection to orignal and migrated schema"
+        ~idx:1
+      |> Test.eval ;
+
+      let query_mainnet_db = Mina_caqti.query mainnet_pool in
+      let query_migrated_db = Mina_caqti.query migrated_pool in
+
+      let%bind end_global_slot =
+        get_migration_end_slot_for_state_hash ~query_mainnet_db fork_state_hash
+      in
+
+      let%bind fork_block_height =
+        query_migrated_db ~f:(fun db ->
+            Sql.Mainnet.blockchain_length_for_state_hash db fork_state_hash )
```

Comment:
It's a little confusing here: you are querying against the migrated db using `Sql.Mainnet`